// @nrwl/react/src/utils/versions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.babelPluginStyledComponentsVersion = exports.eslintPluginReactHooksVersion = exports.eslintPluginReactVersion = exports.eslintPluginJsxA11yVersion = exports.eslintPluginImportVersion = exports.reactTestRendererVersion = exports.reactReduxVersion = exports.reduxjsToolkitVersion = exports.testingLibraryReactVersion = exports.typesReactRouterDomVersion = exports.reactRouterDomVersion = exports.styledJsxVersion = exports.emotionBabelPlugin = exports.emotionReactVersion = exports.emotionStyledVersion = exports.typesStyledComponentsVersion = exports.styledComponentsVersion = exports.typesNodeVersion = exports.typesReactIsVersion = exports.typesReactDomVersion = exports.typesReactVersion = exports.reactIsVersion = exports.reactDomVersion = exports.reactVersion = exports.nxVersion = void 0;
exports.nxVersion = require('../../package.json').version;
exports.reactVersion = '18.2.0';
exports.reactDomVersion = '18.2.0';
exports.reactIsVersion = '18.2.0';
exports.typesReactVersion = '18.0.13';
exports.typesReactDomVersion = '18.0.5';
exports.typesReactIsVersion = '17.0.3';
exports.typesNodeVersion = '18.0.0';
exports.styledComponentsVersion = '5.3.5';
exports.typesStyledComponentsVersion = '5.1.25';
exports.emotionStyledVersion = '11.9.3';
exports.emotionReactVersion = '11.9.3';
exports.emotionBabelPlugin = '11.9.2';
// WARNING: This needs to be in sync with Next.js' dependency or else there might be issues.
exports.styledJsxVersion = '5.0.2';
exports.reactRouterDomVersion = '6.3.0';
exports.typesReactRouterDomVersion = '5.3.3';
exports.testingLibraryReactVersion = '13.3.0';
exports.reduxjsToolkitVersion = '1.8.2';
exports.reactReduxVersion = '8.0.2';
exports.reactTestRendererVersion = '18.2.0';
exports.eslintPluginImportVersion = '2.26.0';
exports.eslintPluginJsxA11yVersion = '6.5.1';
exports.eslintPluginReactVersion = '7.30.0';
exports.eslintPluginReactHooksVersion = '4.6.0';
exports.babelPluginStyledComponentsVersion = '1.10.7';
